README.md

```diff
@@ -1,169 +1,97 @@
 ```
+# Build the tool
+$ make vendor all
 
-### Build Kubernetes
+# Get the recommended container image
+$ sudo machinectl pull-raw --verify=no https://stable.release.core-os.net/amd64-usr/current/coreos_developer_container.bin.bz2 coreos
 
-kube-spawn needs the built Kubernetes binaries and hyperkube
-Docker image. You need to build them like that:
+# Spawn and provision nodes for the cluster
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kube-spawn --kubernetes-version=1.7.0 up --image=coreos --nodes=3
 
-```
-cd $GOPATH/src/k8s.io/kubernetes
-build/run.sh make
-cd cluster/images/hyperkube
-make VERSION=latest
+# Setup Kubernetes
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kube-spawn --kubernetes-version=1.7.0 init
 ```
 
-### Get CNI plugins
+## Run from local Kubernetes repository
 
-```
-go get -u github.com/containernetworking/plugins/plugins/main/bridge
-go get -u github.com/containernetworking/plugins/plugins/ipam/host-local
-```
+The requirements stay the same as above.
 
-`kube-spawn` will configure the networks it needs in `/etc/cni/net.d`.
-
-## Requirements
+To get the Hyperkube image of your local Kubernetes build to deploy, `kube-spawn` sets up
+a local insecure Docker registry. Pushing images to it needs to be enabled by adding
+`10.0.0.1:5000` to the docker daemons config file (`/etc/docker/daemon.json`).
 
-### on the host
+The following steps assume you have a local checkout of the Kubernetes source.
 
-  * systemd-nspawn v233, or systemd-nspawn v231 with backports for:
-    * `SYSTEMD_NSPAWN_USE_CGNS` https://github.com/systemd/systemd/pull/3809
-    * `SYSTEMD_NSPAWN_MOUNT_RW` and `SYSTEMD_NSPAWN_USE_NETNS` https://github.com/systemd/systemd/pull/4395
-  * glide from https://github.com/Masterminds/glide
+```
+# Build Kubernetes
+$ cd $GOPATH/src/k8s.io/kubernetes
+$ build/run.sh make
 
-## Build and run kube-spawn
+# Build a Hyperkube image
+$ cd cluster/images/hyperkube
+$ make VERSION=latest
+```
 
-Make sure you have `glide` available in you PATH.
-In the directory where you cloned this repository, please do:
+Assuming you have built `kube-spawn` and pulled the CoreOS image, do:
 
 ```
-make vendor all
-sudo machinectl pull-raw --verify=no https://stable.release.core-os.net/amd64-usr/current/coreos_developer_container.bin.bz2 coreos
-sudo GOPATH=$GOPATH SYSTEMD_NSPAWN_PATH=<path_to_your_nspawn_binary> CNI_PATH=<path_to_cni_plugins_bins> ./kube-spawn up --image coreos --nodes 2
-sudo ./kube-spawn init
+# Spawn and provision nodes for the cluster
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kube-spawn up --image=coreos --nodes=3
+
+# Setup Kubernetes
+$ sudo GOPATH=$GOPATH CNI_PATH=$GOPATH/bin ./kube-spawn init
 ```
 
-Sometimes when Docker doesn't use the newest existing API, you may see
-the following error:
+## CLI
 
 ```
-2017/01/26 16:41:38 Error when pushing image: Error response from daemon: client is newer than server (client API version: 1.26, server API version: 1.24)
+Available Commands:
```

Comment:
I would just refer to `./kube-spawn -h` instead of duplicating the help output (and having to keep it up-to-date).